lib/index.js

```diff
@@ -85,7 +86,10 @@ class ReactRefreshPlugin {
     }
 
     // Inject react-refresh context to all Webpack entry points
-    compiler.options.entry = injectRefreshEntry(compiler.options.entry, this.options, compiler);
+    compiler.options.entry = injectRefreshEntry(
+      compiler.options.entry,
+      ...getOverlayEntry(this.options, compiler.options.devServer || {})
```

Comment:
this function returns an array so that we can spread it here and pass `prependEntries` and `overlayEntries` as normal function parameters.

Reply:
Initially I was thinking to pass down an object like this:

```js
{
  prepend: ["entries"],
  overlay: ["entries"]
}
```

So then we would only have to pass down one single argument during recursion.

How does this sound to you? I think either way forward is fine.